api/src/middleware/authenticate.ts

```diff
@@ -53,9 +69,15 @@ const authenticate: RequestHandler = asyncHandler(async (req, res, next) => {
 		}
 
 		req.accountability.user = payload.id;
-		req.accountability.role = user.role;
+		req.accountability.user_role = user.role;
 		req.accountability.admin = user.admin_access === true || user.admin_access == 1;
 		req.accountability.app = user.app_access === true || user.app_access == 1;
+
+		if (!req.accountability.admin && req.accountability.maintenance_role) {
+			req.accountability.role = req.accountability.maintenance_role;
+		} else {
+			req.accountability.role = req.accountability.user_role;
+		}
```

Comment:
I think an easier solution here would be to "just" add a:

```js
if (settings.maintenance_enabled) {
  req.accountability.role = maintenance_role;
}
```

there's no need to maintain the original user's role at this point, as we're forcing their role to be overridden